src/target/adiv5.c

```diff
@@ -848,8 +846,7 @@ void adiv5_dp_init(ADIv5_DP_t *dp, const uint32_t idcode)
 				dp->ap_cleanup(i);
 #endif
 			if (++invalid_aps == 8) {
-				adiv5_dp_unref(dp);
-				return;
+				break;
```

Comment:
It would be helpful to include a comment here with where this jumps to (for loop on line 874) and why. Additionally, remove the now defunct {}'s please.

Reply:
Comment added!